packages/contracts/contracts/core/plugin/AragonPlugin.sol

```diff
@@ -3,30 +3,37 @@
 pragma solidity 0.8.10;
 
 import "@openzeppelin/contracts/utils/Context.sol";
-import "@openzeppelin/contracts/proxy/utils/Initializable.sol"; 
+import "@openzeppelin/contracts/proxy/utils/Initializable.sol";
+import "@openzeppelin/contracts/utils/introspection/ERC165.sol";
 
-import { PermissionManager } from "../permission/PermissionManager.sol";
+import {PermissionManager} from "../permission/PermissionManager.sol";
 
 import "../../utils/AppStorage.sol";
 import "../IDAO.sol";
 
 /// @title AragonPlugin
 /// @notice An abstract contract to inherit from when creating a non-upgradable plugin.
-abstract contract AragonPlugin is Initializable, AppStorage, Context {
-
+abstract contract AragonPlugin is Initializable, ERC165, AppStorage, Context {
```

Comment:
Instead of creating a second version of `Component` with a different name, we should rename `Component` to `Plugin` and refactor it into the state we want it to be in.